build.rs

```diff
@@ -0,0 +1,27 @@
+use std::process::Command;
+use std::str;
+
+fn main() {
+    // Temporary check to see if the rustc version >= 1.81 in which case the
+    // `Error` trait is always available. This is temporary because in the
+    // future the MSRV of this crate will be beyond 1.81 in which case this
+    // build script can be deleted.
+    let minor = rustc_minor_version().unwrap_or(0);
+    if minor >= 81 {
+        println!("cargo:rustc-cfg=core_error");
+    }
+    if minor >= 80 {
+        println!("cargo:rustc-check-cfg=cfg(core_error)");
```

Comment:
A note for my understanding, this is because of https://blog.rust-lang.org/2024/05/06/check-cfg.html#custom-cfgs-in-build-scripts